File: backend/src/main/java/ca/metricalsky/winston/entity/ThumbnailEntity.java
package ca.metricalsky.winston.entity;

import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "thumbnails")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ThumbnailEntity {

    @Id
    @Column(name = "id")
    private String id;

    @Basic(optional = false)
    @Column(name = "url")
    private String url;

    @Basic(optional = false)
    @Column(name = "image")
    private byte[] image;
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/entity/ThumbnailEntity.kt
package ca.metricalsky.winston.database.entity

import jakarta.persistence.Basic
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "thumbnails")
class ThumbnailEntity(

    @Id
    @Column(name = "id")
    var id: String,

    @Basic(optional = false)
    @Column(name = "url")
    var url: String,

    @Basic(optional = false)
    @Column(name = "image")
    var image: ByteArray
)
